code/game/objects/items/toys.dm

```diff
@@ -127,6 +127,73 @@
 	throw_range = 7
 	force = 0
 	var/random_color = TRUE
+	var/world_state
+	var/storage_state
+	var/current_color
+
+/obj/item/toy/balloon/long
+	name = "long balloon"
+	desc = "A perfect balloon to contort into goofy forms."
+	icon_state = "balloon_long"
+	inhand_icon_state = "balloon"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/toy/balloon/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(isturf(loc))
+		icon_state = "[world_state]"
+	else
+		icon_state = "[storage_state]"
+	update_appearance()
+
+/obj/item/toy/balloon/long/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(istype(attacking_item, /obj/item/toy/balloon/long) && HAS_TRAIT(user, TRAIT_BALLOON_SUTRA))
+		var/obj/item/toy/balloon/long/hit_by = attacking_item
+		if(hit_by.current_color != current_color)
```

Comment:
holy fucking shit this if chain is absolutely horrific. here's how we can make it better. first, early returns to make the indentation less bad

```dm
/obj/item/toy/balloon/long/balloon/attackby(...)
 if(!istype(attacking_item, /obj/item/toy/balloon/long) || !HAS_TRAIT(user, TRAIT_BALLOON_SUTRA)
  return ..() // saves indentation while doing the exact same thing.
 
 var/obj/item/toy/balloon/long/hit_by = attacking_item
 if(hit_by.current_color == current_color)
  to_chat(user, span_warning("You must use balloons of different colours to do that!"))
  return ..()

 // then you would have the rest of the code (visible_message, etc.) here on this indent line and qdel both items and return TRUE at the very end.
```

we'll need to do more work, but start here. you would de-indent all the coming lines to just make it slightly less insane.